core/matchers/matcher.ts

```diff
@@ -105,6 +110,20 @@ export class Matcher<T> {
     }
   }
 
+  private _checkBuffersAreEqual(_bufferA: any, _bufferB: any): boolean {
+    try {
+      const bufferA = Buffer.from(_bufferA);
```

Comment:
If `_bufferA` or `_bufferB` are already instances of `Buffer`, this will clone them - could be intensive. Also, we know that `Buffer.isBuffer` is `true` so there's no need to convert.

Reply:
I could change these to `const bufferA = Buffer.isBuffer(_bufferA) ? _bufferA : Buffer.from(_bufferA)`? See response to your other comment for why I am converting to new buffers.

Reply:
Updated as above.